### PR TITLE
Replace old HotRod index annotation with new one

### LINKED ISSUE #10988: Remove doubled field from HotRod entities
### Description

Currently, for case-insensitive search and order we have some fields doubled, e.g.:
https://github.com/keycloak/keycloak/blob/19e55650a20592e843485cc79dc648ddea24a56c/model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java#L64-L70

The cause is that by using the analyzed index, we achieve case-insensitivity but we lose the possibility to order because the analyzed index doesn't know the original value.

We should be able to replace the doubled field with a doubled index (we should be able to use two annotations on the same field) which should end up with fewer data in the store. 
 

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### LINKED ISSUE #12493: Replace old HotRod index annotation with new one
Infinispan introduced new annotations in https://issues.redhat.com/browse/ISPN-13842, we should use these instead of the old one

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #12493

### CODE REVIEW

**Review by mhajas — CHANGES_REQUESTED** [review]
Thank you for the PR @martin-kanis. I added two comments that I would say are worth attention. Could you please have a look?

Also, were you able to play with no-downtime upgrades for these new annotations? For example, from sortable to non-sortable, or projectable and so on?

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authSession/HotRodRootAuthenticationSessionEntity.java [hunk 1]
```diff
@@ -62,21 +64,21 @@ public interface HotRodRootAuthenticationSessionEntitySchema extends GeneratedSc
     }
 
 
-    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @Basic(sortable = true)
```
[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authSession/HotRodRootAuthenticationSessionEntity.java]: Changed to `projectable` after the performance analyses.

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryOperators.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ private static String notExists(String modelFieldName, Object[] values, Map<Stri
     private static String iLike(String modelFieldName, Object[] values, Map<String, Object> parameters) {
         String sanitizedValue = (String) IckleQueryMapModelCriteriaBuilder.sanitizeNonAnalyzed(values[0]);
         return singleValueOperator(ModelCriteriaBuilder.Operator.ILIKE)
-                .combine(modelFieldName + "Lowercase", new String[] {sanitizedValue.toLowerCase()}, parameters);
+                .combine(modelFieldName, new String[] {sanitizedValue.toLowerCase()}, parameters);
```
[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryOperators.java]: I added those checks and removed `toLowerCase()` by simplifying `ilike()` and `like()` methods into an one method.

**Review by mhajas — APPROVED** [review]
Thanks @martin-kanis! The changes LGTM, I added just one question that should not block merging. There are some failures in the GHA (probably not caused by this), could you please rebase?

**Review by mhajas — CHANGES_REQUESTED** [review]
@martin-kanis Can you please rebase to have GHA running with hotrod?

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authSession/HotRodRootAuthenticationSessionEntity.java [hunk 3]
```diff
@@ -62,21 +64,21 @@ public interface HotRodRootAuthenticationSessionEntitySchema extends GeneratedSc
     }
 
 
-    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
+    @Basic(projectable = true)
     @ProtoField(number = 1)
     public Integer entityVersion = VERSION;
 
     @ProtoField(number = 2)
     public String id;
```
[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authSession/HotRodRootAuthenticationSessionEntity.java:72]: Yes, it is possible. I tried it in my [PR](https://github.com/infinispan/infinispan/pull/10545) but I didn't commit it.

Also it is possible to project fields that are indexed, but not `projectable`. https://github.com/infinispan/infinispan/pull/10545/files#diff-a3580c88f69f3adefb7c2580e2aa7abcb0fe6e4f56d9058e7fd4ec35274af7c5R458

[the remark this answers is not in the document]
Comment (mhajas) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/authSession/HotRodRootAuthenticationSessionEntity.java:72]: ok, thanks for the confirmation @martin-kanis!

**Comment by martin-kanis** [comment]
@mhajas All good after rebase.

**Comment by mhajas** [comment]
> @mhajas All good after rebase.

I will just test locally, to be 100% sure the integration tests are running with the new annotations and if it is ok, I will merge.

**Review by mhajas — APPROVED** [review]
Thanks @martin-kanis, this works correctly. I am merging.

**Pull request merged**